core/src/main/java/jenkins/util/JSONSignatureValidator.java

```diff
@@ -73,7 +75,17 @@ public FormValidation verifySignature(JSONObject o) throws IOException {
                     certs.add(c);
                 }
 
-                CertificateUtil.validatePath(certs, loadTrustAnchors(cf));
+                try {
+                    CertificateUtil.validatePath(certs, loadTrustAnchors(cf));
+                } catch (GeneralSecurityException e) {
+                    Iterable<Object> certSubjectDNs = Iterables.transform(certs, new Function<X509Certificate, Object>() {
+                        @Override
+                        public Object apply(X509Certificate cert) {
+                            return cert == null ? null : cert.getSubjectDN();
+                        }
+                    });
+                    return FormValidation.warning(e,String.format("Certificate validation failed with certs %s for %s: %s", certSubjectDNs.toString(), name, e.toString()));
```

Comment:
Isn't one of the root causes here that the earlier warnings should be returned to the user and they where not as validatePath then failed?

(see lines 70,72)
The message as returned is pretty nasty to comprehend and will swallow the more human parsable message from the mentioned lines.